src/Uplink/Auth/Token/Token_Manager.php

```diff
@@ -59,44 +61,116 @@ public function validate( string $token ): bool {
 	/**
 	 * Store the token.
 	 *
+	 * @since TBD Added $slug param.
+	 *
 	 * @param  string  $token
+	 * @param  string  $slug The Product slug to store the token for.
 	 *
 	 * @return bool
 	 */
-	public function store( string $token ): bool {
+	public function store( string $token, string $slug = '' ): bool {
 		if ( ! $token ) {
 			return false;
 		}
 
+		$slug = $this->validate_slug( $slug );
+
+		$current_value = $this->get( $slug );
+
 		// WordPress would otherwise return false if the items match.
-		if ( $token === $this->get() ) {
+		if ( $token === $current_value ) {
 			return true;
 		}
 
-		return update_network_option( get_current_network_id(), $this->option_name, $token );
+		$data = $this->get_all();
+
+		if ( ! is_array( $data ) ) {
+			$data = [];
+		}
+
+		$data[ $slug ] = $token;
+
+		return update_network_option( get_current_network_id(), $this->option_name, $data );
 	}
 
 	/**
 	 * Get the token.
 	 *
+	 * @since TBD Added $slug param. Changed return type to mixed.
+	 *
+	 * @param  string  $slug The Product slug to retrieve the token for.
+	 *
 	 * @return string|null
 	 */
-	public function get(): ?string {
-		return get_network_option( get_current_network_id(), $this->option_name, null );
+	public function get( string $slug = '' ): ?string {
+		$slug = $this->validate_slug( $slug );
+
+		$value = get_network_option( get_current_network_id(), $this->option_name, null );
+
+		if ( ! $value ) {
+			return null;
+		}
+
+		if ( ! $slug ) {
+			return is_string( $value ) ? $value : ( array_values( $value )[0] ?? null );
+		}
+
+		if ( is_string( $value ) ) {
+			// Still using old structure, lets return whatever we found.
+			return $value;
+		}
+
+		return $value[ $slug ] ?? null;
+	}
+
+	/**
+	 * Get all the tokens.
+	 *
+	 * @since TBD
+	 *
+	 * @return null|string|array
+	 */
+	public function get_all() {
+		return get_network_option( get_current_network_id(), $this->option_name, [] );
 	}
 
 	/**
 	 * Revoke the token.
 	 *
 	 * @return bool
 	 */
-	public function delete(): bool {
+	public function delete( string $slug = '' ): bool {
+		$current_value = $this->get_all();
 		// Already doesn't exist, WordPress would normally return false.
-		if ( $this->get() === null ) {
+		if ( $current_value === null ) {
+			return true;
+		}
+
+		$slug = $this->validate_slug( $slug );
+
+		if ( ! $slug ) {
+			return delete_network_option( get_current_network_id(), $this->option_name );
```

Comment:
Is this saying that if we don't have the slug in the resource collection, we delete all the data?

Wouldn't that delete other slug data as well or am I missing something?